src/iff.imageio/iffinput.cpp

```diff
@@ -713,23 +787,87 @@ IffInput::readimg()
                 return false;
             }
 
-            // tile
-            t++;
+            rgbatiles++;
+
+            // chunk type: ZBUF
+        } else if (std::memcmp(chunktype, iff_zbuf_tag, 4) == 0) {
+            // get tile coordinates.
+            uint16_t xmin, xmax, ymin, ymax;
+            if (!read(&xmin) || !read(&ymin) || !read(&xmax) || !read(&ymax)) {
+                errorfmt("IFF error io read xmin, ymin, xmax and ymax failed");
+                return false;
+            }
+
+            // get tile width/height
+            uint32_t tw = xmax - xmin + 1;
+            uint32_t th = ymax - ymin + 1;
+
+            // get image size
+            // skip coordinates, uint16_t (2) * 4 = 8
+            uint32_t image_size = chunksize - 8;
+
+            // check tile
+            if (xmin > xmax || ymin > ymax || xmax >= m_spec.width
+                || ymax >= m_spec.height || !tw || !th) {
+                errorfmt(
+                    "IFF error io xmin, ymin, xmax or ymax does not match");
+                return false;
+            }
+
+            // set tile size
+            std::vector<uint8_t> scratch;
+
+            // set bytes.
+            scratch.resize(image_size);
+
+            // zbuffer is always compressed in IFF
+
+            if (!ioread(scratch.data(), 1, scratch.size())) {
+                errorfmt("IFF error io seek failed");
+                return false;
+            }
+
+            // set tile data
+            uint8_t* p = scratch.data();
+
+            for (int c = m_header.zbuffer_bytes() - 1; c >= 0; --c) {
+                std::vector<uint8_t> in(tw * th);
+                uint8_t* in_p = in.data();
+
+                // uncompress and increment
+                p += uncompress_rle_channel(p, in_p, tw * th);
+
+                // set tile
+                for (uint32_t py = ymin; py <= ymax; py++) {
+                    uint8_t* out_dy = static_cast<uint8_t*>(m_buf.data())
+                                      + (py * m_header.width)
+                                            * m_header.pixel_bytes();
+
+                    for (uint16_t px = xmin; px <= xmax; px++) {
+                        uint8_t* out_p = out_dy + px * m_header.pixel_bytes()
+                                         + m_header.rgba_channels_bytes() + c;
+                        *out_p++ = *in_p++;
+                    }
+                }
+            }
+            ztiles++;
 
         } else {
             // skip to the next block
-            if (!ioseek(chunksize))
+            if (!ioseek(chunksize, SEEK_CUR)) {
+                errorfmt("IFF error io seek failed");
```

Comment:
Not necessary -- the ioseek method itself calls errorfmt if it fails. All you have to do here is return.